docs/did/did-crypto-assumptions.md

```diff
@@ -0,0 +1,221 @@
+
+## Creating a Wallet
+
+A wallet is a json structure which holds keys. Wallets are protected with a password, key derivation function and symmetric encryption.
+
+We generate a symmetric key using a public salt and the password provided by the user:
+
+```
+sodium.to_hex(
+    sodium.crypto_pwhash(
+      sodium.crypto_box_SEEDBYTES,
+      password,
+      sodium.from_hex(salt),
+      sodium.crypto_pwhash_OPSLIMIT_MIN,
+      sodium.crypto_pwhash_MEMLIMIT_MIN,
+      sodium.crypto_pwhash_ALG_DEFAULT,
+    ),
+  );
+```
+
+[Read More](https://libsodium.gitbook.io/doc/password_hashing)
+[Our Code](https://github.com/transmute-industries/transmute/blob/master/packages/transmute-did/src/lib/cryptoSuites/sodiumExtensions/index.js#L66)
+
+We use this key to encrypt the wallet json which contains private keys, symmetric keys, key shares, or other senstive information which should never be written to disk in plaintext.
+
+```
+const nonce = sodium.randombytes_buf(sodium.crypto_secretbox_NONCEBYTES);
+  const encrypted = sodium.crypto_secretbox_easy(
+    data,
+    nonce,
+    sodium.from_hex(key),
+  );
+```
+
+[Read More](https://libsodium.gitbook.io/doc/secret-key_cryptography)
+[Our Code](https://github.com/transmute-industries/transmute/blob/master/packages/transmute-did/src/lib/cryptoSuites/sodiumExtensions/index.js#L88)
+
+Wallets can be decrypted, when access to keys is required. They should never be stored in plaintext.
+
+## Create a DID Document
+
+> Decentralized Identifiers (DIDs) are a new type of identifier for verifiable, "self-sovereign" digital identity. DIDs are fully under the control of the DID subject, independent from any centralized registry, identity provider, or certificate authority. DIDs are URLs that relate a DID subject to means for trustable interactions with that subject. DIDs resolve to DID Documents — simple documents that describe how to use that specific DID. Each DID Document contains at least three things: cryptographic material, authentication suites, and service endpoints. Cryptographic material combined with authentication suites provide a set of mechanisms to authenticate as the DID subject (e.g., public keys, pseudonymous biometric protocols, etc.). Service endpoints enable trusted interactions with the DID subject.
+>
+> This document specifies a common data model, format, and operations that all DIDs support.
+
+[Read More](https://w3c-ccg.github.io/did-spec/)
+
+Transmute DID does not have opinions about where you store your DID Document, what technologies you use to do so, etc. In order to use `transmute-did` to build a valid DID Method, you MUST ensure that a document resolved from an identifier has the corrent id property. `transmute-did` does support signing the DID Document as described in the spec. See [binding-of-identity](https://w3c-ccg.github.io/did-spec/#binding-of-identity).
+
+Here is a DID Document produced from `transmute-did` that uses OrbitDB to help with document resolution and storage:
```

Comment:
Maybe include a reference to this https://github.com/transmute-industries/transmute/blob/af5e1db73fbc39a8d437cc0d5ea8b025ee42afb5/docs/did/nested-did-spec.md somewhere around here ?